src/firmware_new/boards/frankie_v1/io/drivetrain.c

```diff
@@ -0,0 +1,48 @@
+#include "firmware_new/boards/frankie_v1/io/drivetrain.h"
+
+#include <assert.h>
+#include <stdbool.h>
+
+static DriveTrainUnit_t *_front_left_drive_unit;
+static DriveTrainUnit_t *_front_right_drive_unit;
+static DriveTrainUnit_t *_back_left_drive_unit;
+static DriveTrainUnit_t *_back_right_drive_unit;
+
+static bool initialized = false;
+
+void io_drivetrain_init(DriveTrainUnit_t *front_left_drive_unit,
+                        DriveTrainUnit_t *front_right_drive_unit,
+                        DriveTrainUnit_t *back_left_drive_unit,
+                        DriveTrainUnit_t *back_right_drive_unit)
+{
+    _front_left_drive_unit  = front_left_drive_unit;
+    _front_right_drive_unit = front_right_drive_unit;
+    _back_left_drive_unit   = back_left_drive_unit;
+    _back_right_drive_unit  = back_right_drive_unit;
+
+    initialized = true;
+}
+
+void io_drivetrain_applyForceFrontLeftWheel(float force_newtons)
+{
+    assert(initialized);
```

Comment:
Any reason why each wheel has it's own function when the drive unit could be a parameter instead?

Reply:
Great question! We need a `static` function for each of the wheels so that when we call `App_Wheel_create` we can pass in the `apply_force` function.